### PR TITLE
React capstone project - Metrics webapp

### PR DESCRIPTION
**Pull request by sam2if**
**Project requirements**
API First you need to choose an API to base the development of the webapp on. The API should allow you to get numeric values based on a parameter. We recommend that you choose an API that is open or needs an API key. If you choose an API that require authentication, you should implement it on your own.

**Some example APIs are:**

[Financial modeling prep API](https://financialmodelingprep.com/developer/docs/): stock data.
[Air pollution API](https://openweathermap.org/api/air-pollution): air quality data.
You can find many APIs in [this GitHub repo](https://github.com/public-apis/public-apis). Some of the APIs requires a token for authentication and some others are just open.

**Design**

You should follow these [design guidelines](https://www.behance.net/gallery/31579789/Ballhead-App-(Free-PSDs)), including:
Colors (select one main color for your website).
Typography: font face, size and weight.
Layout: composition and space between elements.
Original design idea by [Nelson Sakwa on Behance](https://www.behance.net/sakwadesignstudio).

The [Creative Commons license of the design](https://creativecommons.org/licenses/by-nc/4.0/) requires that you give appropriate credit to the author. Therefore, you must do it in the README of your project.

**Interactions**

**Home page**

When the page loads, the webapp shows the list of categories that could be filtered out by some parameter, for example by the category name.
Along with the category name, you will display some numeric values per category that come from the API.
When the user clicks (or taps) on a category item, the application navigates to the details page.
**Details page**

In the details page, the webapp retrieves data from the API to show detailed data in the category.
When the user clicks on the "Back" button (<), the user navigates to the home page.
Testing requirements

Create unit tests for pure functions in your Redux code.
Create integration tests for your application using the React Testing Library.
You may need to mock the access to the API, so that your tests don't send actual requests.
You may need to mock the connection to the Redux Store.
**Technical requirements**

The project is a single-page application (SPA) built with React and Redux.
The data retrieved from the API should be stored in the Redux store.
You should filter the data that you retrieve from the API using a Filter stateless component.
Every page (the main page and the pages for each item) should have a unique route within the SPA.
The project should be deployed and accessible online using [Render](https://render.com/), [Netlify](https://www.netlify.com/) or [GitHub Pages](https://pages.github.com/).
Add a link to the online version of your app to your README file.

### CODE REVIEW

File: src/_test_/Currency.test.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    };
+    const mytree = render(
+      <MemoryRouter>
+        <Provider store={store}>
+          <Currency eachCurrency={data} />
+        </Provider>
+      </MemoryRouter>,
+    );
+    expect(mytree).toMatchSnapshot();
+  });
```
Comment (ichala) [inline src/_test_/Currency.test.js:31]: - [ ] Good job creating some unit tests , but as part of the requirements kindly create some Unit tests as well 🙏

**Review by ichala — CHANGES_REQUESTED** [review]
Hi Student
Good job so far! 
👏👏👏 
You still need to work on some issues to go to the next project, but you're almost there!

the-simpsons-homer-simpson

## To Highlight 🎉

- No Linter errors  ✔️ 
- Used Redux for state management ✔️
- Good Design ✔️


 ## Required Changes ♻️

- [ ] 🛑 As part of the requirements kindly make sure to record a video presenting the features of the project:
- The maximum length of the video is 5 minutes.
- The video contains a clear and objective description of the project in professional English. The - topic and purpose of the project are made clear.
- The video contains a demo of the project, highlighting all the project features.
- The video contains a highlight about a piece of the code the student finds interesting or is proud of making.

- [ ]  **_🛑 As part of the requirements kindly make sure to deploy your app using Render, Netlify or GitHub Pages , The project should be deployed properly, the home and detail page works as expected_**
- [ ] **_🛑Kindly make sure to attach the video presentation link in your PR and Readme file as well as the deployment link_**

- [ ] **_🛑  Check The Comments Below_**





## Optional suggestions

- N/A

**********
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md [hunk 2]
```diff
@@ -1 +1,204 @@
...
+
+## 💻 Getting Started <a name="getting-started"></a>
+
+- To get a local copy up and running, follow these steps.
+
+<!-- LIVE Video -->
+
+## 🎥 Live Video <a name="live-video"></a>
+
+- [Live video Link](https://github.com/sam2if/REACT_CAPSTONE)
```
Comment (Felix45) [inline README.md]: - [ Required ] Good job working on some of the changes requested by the previous reviewer, however, you forgot to include the correct link to the presentation video on your readme file. Currently, when I click on this link it redirects to the repository and not the presentation video. Kindly rectify this.

- [ Required ] Please note the presentation video should cover the following areas as had been requested by the previous reviewer.
  - The maximum length of the video is 5 minutes.
  - The video contains a clear and objective description of the project in professional English. The - topic and purpose of the project are made clear.
  - The video contains a demo of the project, highlighting all the project features.
  - The video contains a highlight about a piece of the code the student finds interesting or is proud of making.

## Project requirements
Screenshot from 2023-08-10 17-53-20

**Review by Felix45 — CHANGES_REQUESTED** [review]
Hi @sam2if, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.



 ## Highlights :heavy_check_mark: 
- [x] Good UI design :white_check_mark: 
- [x] Application works as expected :white_check_mark: 
- [x] `Redux` state management library is used to manage the state of the application :white_check_mark:  
- [x] Git flow is used correctly :white_check_mark:  
- [x] Implemented some of the changes from the previous reviewer :white_check_mark:   


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md [hunk 3]
```diff
@@ -1 +1,204 @@
...
+## 💻 Getting Started <a name="getting-started"></a>
+
+- To get a local copy up and running, follow these steps.
+
+<!-- LIVE Video -->
+
+## 🎥 Live Video <a name="live-video"></a>
+
+- [Live video Link](https://www.loom.com/share/459edb0dcea34e6d92dd284e3b22c6c6?sid=f38439e3-87f9-4c98-ad3a-776de6fee35a)
+
```
Comment (aamirkhan2478) [inline README.md]: - [ ] As mentioned by the previous reviewer :point_up_2:  the video highlights a piece of the code the student finds attractive or is proud of making but in your video, you didn't highlight the code which is proud of making so kindly re-create your video and also add the piece of code is proud of making :slightly_smiling_face:

**Review by aamirkhan2478 — CHANGES_REQUESTED** [review]
Hi @sam2if, 



Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't been addressed yet.

 ## Highlights
✔️ Good Effort
✔️ Your design is good.
✔️ All linters passed

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by ShadyShawkat — APPROVED** [review]
## Hi @sam2if, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight
- Correct use of GitFlow ✔️ 
- Descriptive commit messages ✔️ 
- Professional looking pull request ✔️ 
- Your readme looks professional ✔️ 

Image

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**